src/test/java/lambda/part2/exercise/Exercise1.java

```diff
@@ -0,0 +1,67 @@
+package lambda.part2.exercise;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import lambda.data.Person;
+import org.junit.Test;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+public class Exercise1 {
+
+    @Test
+    public void ageExtractorFromPersonUsingMethodReference() {
+        Person person = new Person("Иван", "Мельников", 33);
+
+        // TODO создать переменную ageExtractor: Person -> Integer, используя Function и ссылку на метод
+        Function<Person, Integer> ageExtractor=Person::getAge;
+
+        assertEquals(33, ageExtractor.apply(person).intValue());
+
+    }
+
+    @Test
+    public void sameAgesCheckerUsingBiPredicate() {
+        Person person1 = new Person("Иван", "Мельников", 33);
+        Person person2 = new Person("Дмитрий", "Гущин", 33);
+        Person person3 = new Person("Илья", "Жирков", 22);
+
+        // TODO создать переменную sameAgesChecker: (Person, Person) -> boolean, используя BiPredicate
+
+        BiPredicate<Person, Person> sameAgesChecker = (p1,p2)->p1.getAge()==p2.getAge();
+         assertTrue(sameAgesChecker.test(person1, person2));
+         assertFalse(sameAgesChecker.test(person1, person3));
+         assertFalse(sameAgesChecker.test(person2, person3));
+
+        }
+
+    // TODO метод getFullName: Person -> String, извлекающий из объекта Person строку в формате "имя фамилия".
+     private static String getFullName(Person p) {return p.getFirstName()+" "+p.getLastName();}
+
+    // TODO метод createExtractorAgeOfPersonWithTheLongestFullName: (Person -> String) -> ((Person, Person) -> int),
+    // TODO - принимающий способ извлечения полного имени из объекта Person
+    // TODO - возвращающий BiFunction, сравнивающий два объекта Person и возвращающий возраст того, чье полное имя длиннее.
+     public BiFunction<Person,Person, Integer> createExtractorAgeOfPersonWithTheLongestFullName (Function<Person, String> function){
+        return (p1, p2)->p1.getAge()>p2.getAge()?p1.getAge():p2.getAge();
```

Comment:
You must compare persons by full name and return age of person with longest name.